applications/CompressiblePotentialFlowApplication/custom_elements/adjoint_base_potential_flow_element.cpp

```diff
@@ -35,7 +35,8 @@ namespace Kratos
     template <class TPrimalElement>
     void AdjointBasePotentialFlowElement<TPrimalElement>::InitializeSolutionStep(ProcessInfo& rCurrentProcessInfo)
     {
-        mpPrimalElement->Data() = this->Data();
```

Comment:
Adjoint elements have a pointer to its primal element, and when we initialize the adjoint elements we copy its data to the primal element that they have stored. I have updated this part using the new methods

Reply:
This was copying what is done in the Structural app, where this operation is performed at a custom process:
https://github.com/KratosMultiphysics/Kratos/blob/f4f9e45e7adbccff85c1ae77da6a63c9279638b4/applications/StructuralMechanicsApplication/custom_response_functions/adjoint_processes/replace_multiple_elements_and_conditions_process.cpp#L79-L81
Maybe this new data container change is also affecting here? @armingeiser @MFusseder

Reply:
Please note that the flags will work the same. (There are different copies in each condition and element sharing a geometry)